internal/cmd/agent/controller/bundledeployment_controller.go

```diff
@@ -143,16 +142,10 @@ func (r *BundleDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			}
 		}
 
-		// TODO is this needed with drift correction?
-		if len(bd.Status.ModifiedStatus) > 0 && monitor.ShouldRedeploy(bd) {
-			result.RequeueAfter = durations.MonitorBundleDelay
-			logger.Info("Redeploying, by resetting the appliedDeploymentID")
+		if len(bd.Status.ModifiedStatus) > 0 && monitor.IsAgent(bd) && monitor.ShouldRedeploy(bd) {
```

Comment:
we don't need `montitor.IsAgent` as it is checked inside [ShouldRedeploy](https://github.com/rancher/fleet/blob/30ca349a84bcb80f6b2bc5eb26f86c098aaf6b58/internal/cmd/agent/deployer/monitor/updatestatus.go#L49)